frontend/src/components/pages/topics/QuickInfo.tsx

```diff
@@ -15,74 +15,104 @@ import { api } from '../../../state/backendApi';
-                    value={formatConfigValue(configEntry!.name, configEntry?.value, 'friendly')}
-                />
-            );
+    const segmentMs = filteredConfigEntries?.find(e => e.name === 'segment.ms')
+    const segmentBytes = filteredConfigEntries?.find(e => e.name === 'segment.bytes')
 
-        if (configStats.length > 0)
-            statsAr.push(<StatsSeparator key={'separator'} />);
-
-        statsAr.push(...configStats);
+    if(!configEntries || !filteredConfigEntries || !cleanupPolicy) {
+        return null
     }
 
-    return <Flex>{statsAr}</Flex>
+    return (
+      <Flex as="dl" gap={4} my={4}>
+          <Flex gap={2}>
+              <Text as="dt" fontWeight="bold">
+                  Size:
+              </Text>
+              <Text as="dd">
+                  {!!topic ? prettyBytesOrNA(topic.logDirSummary.totalSizeBytes):'...'}
+              </Text>
+          </Flex>
+          <Box>
+              <Divider orientation="vertical"/>
+          </Box>
+          <Flex gap={2}>
+              <Tooltip
+                label="The number of messages shown is an estimate. This is calculated by summing the differences between the highest and lowest offsets in each partition. The actual number of messages may vary due to factors such as message deletions, log compaction, and uncommitted or transactional messages."
+                hasArrow
+                placement="bottom"
+              >
+                  <Flex alignItems="flex-end">
+                      <MdInfoOutline size={16}/>
+                  </Flex>
+              </Tooltip>
+              <Text as="dt" fontWeight="bold">
+                  Estimated messages:
+              </Text>
+              <Text as="dd">
+                {messageSum}
+              </Text>
+          </Flex>
+        <Box>
+          <Divider orientation="vertical"/>
+        </Box>
+        {cleanupPolicy && <Flex gap={2}>
+          <Text as="dt" fontWeight="bold">Cleanup Policy:</Text>
+          <Text as="dd">
+            {({
+              'compact': 'Compact',
+              'compact,delete': 'Compact & Delete',
+              'delete': 'Delete',
+            } as Record<CleanupPolicyType, string>)[cleanupPolicy as CleanupPolicyType] ?? ''}
+          </Text>
+        </Flex>}
+        <Box>
+          <Divider orientation="vertical"/>
+        </Box>
+        <Flex gap={2}>
+          {cleanupPolicy==='compact' && <><Text as="dt" fontWeight="bold">Segment:</Text>
+              {segmentMs && segmentBytes && <Text as="dd">~
+                  <>{formatConfigValue(segmentMs.name, segmentMs.value, 'friendly')} or {formatConfigValue(segmentBytes.name, segmentBytes.value, 'friendly')}</>
+              </Text>}
+          </>}
+
+          {cleanupPolicy==='delete' && <><Text as="dt" fontWeight="bold">Retention:</Text>
+            {retentionMs && retentionBytes && <Text as="dd">~
+                {retentionMs.value === '-1' && retentionBytes.value === '-1' ? 'Unlimited':
```

Comment:
@weeco @rikimaru0345 do you know how it's with the type here? TS says it's always string or undefined, but in the ticket it says it's -1 like a number.

Reply:
`retention.ms` is a long with valid values between -1 and max long. The same should apply to retention.bytes

Reply:
I think in that case we need to improve the typings, but it's probably out of scope of this PR